bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,552 @@
+                                    } else if ("12".contentEquals(swingLRString[i1])) {
+                                        swingLRModes.add("WIDE");
+                                    } else {
+                                        swingLRModes.add(swingLRString[i1]);
+                                    }
+                                }
+                                channelId = CHANNEL_TYPE_VANESLR;
+                                properties.put("Supported vane left/right modes", swingLRModes.toString());
+                                updateProperties(properties);
+                                addChannel(channelId, itemType, swingLRModes);
+                                break;
+                            case 9:
+                                logger.trace("UID : {} ; minValue : {}", datapoints[i].uid, descr.minValue);
+                                logger.trace("UID : {} ; maxValue : {}", datapoints[i].uid, descr.maxValue);
+                                channelId = CHANNEL_TYPE_TARGETTEMP;
+                                itemType = "Number:Temperature";
+                                addChannel(channelId, itemType, null);
+                                break;
+                            case 10:
+                                logger.trace("UID : {} ; minValue : {}", datapoints[i].uid, descr.minValue);
+                                logger.trace("UID : {} ; maxValue : {}", datapoints[i].uid, descr.maxValue);
+                                channelId = CHANNEL_TYPE_AMBIENTTEMP;
+                                itemType = "Number:Temperature";
+                                addChannel(channelId, itemType, null);
+                                break;
+                            case 37:
+                                logger.trace("Add Channel Outdoor Temperature");
+                                logger.trace("UID : {} ; description : {}", datapoints[i].uid, datapoints[i].descr);
+                                logger.trace("UID : {} ; minValue : {}", datapoints[i].uid, descr.minValue);
+                                logger.trace("UID : {} ; maxValue : {}", datapoints[i].uid, descr.maxValue);
+                                channelId = CHANNEL_TYPE_OUTDOORTEMP;
+                                itemType = "Number:Temperature";
+                                addChannel(channelId, itemType, null);
+                                break;
+                        }
+                    }
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+            logger.trace("Start Refresh Job");
+            refreshJob = scheduler.scheduleWithFixedDelay(this::getAllUidValues, 0, INTESIS_REFRESH_INTERVAL_SEC,
+                    TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+    }
+
+    @SuppressWarnings("null")
```

Comment:
Please add a null check instead of suppressing the warning.

Reply:
Done